### PR TITLE
Add fade view

### PR DESCRIPTION
**Pull request by arturs105**
Fade-outs are used in multiple places, so i created a view that fades out left and right sides of it's children.

I didn't change existing fades, because the fade widths aren't known yet. I will do it in another PR.

### CODE REVIEW

**Comment by amulware** [comment]
Will this work for views that change their size?

**Comment by arturs105** [comment]
I think it should, since the fade is added in `LayoutSubviews`.

**Comment by amulware** [comment]
I just noticed that it creates a new gradient layer every time, which seems wasteful. First I thought it might be adding more and more of them, but it replaces instead, so it's probably fine for resizing views. Though if the layer could be reused, that would be great.

**Comment by arturs105** [comment]
I modified the `FadeView` so it updates the mask only when the size of it changes. Size changes of child views don't require mask updates.

File: Toggl.Daneel/Views/FadeView.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            base.LayoutSubviews();
+
+            if (fadeLayer?.Bounds == Bounds) return;
+
+            addFading();
+        }
+
+        private void addFading()
+        {
+            fadeLayer = new CAGradientLayer();
```
Comment (amulware) [inline Toggl.Daneel/Views/FadeView.cs]: Why not reuse the same layer, instead of creating multiple ones?

**Pull request merged**